### PR TITLE
[gaudi] new versions 36.1 36.2 36.3

### PR DESCRIPTION
**Pull request by wdconinc**
Quoting [v36r3](https://gitlab.cern.ch/gaudi/Gaudi/-/tags/v36r3): "This is a bugfix release with a couple of changes that technically are backward incompatible
because they fix issues that might have gone unnoticed."

For this reason, I am marking v36r2 as preferred to give dependents an opportunity to test (all dependents are in third-party repositories).

Maintainers tag: @drbenmorgan @vvolkl

### CODE REVIEW

**Review by alalazo — DISMISSED** [review]
LGTM. I'll be giving some time for maintainers to review setting the preferred to v36.2. Feel free to ping me when you want to merge.

**Comment by vvolkl** [comment]
To me the changes don't look like they should cause issues for users of this recipe, so I think you can drop the `preferred`. I'll make a test build for key4hep to be sure.

**Comment by wdconinc** [comment]
I didn't have any issues with 36.3 either. Let me know if you can build your stack succesfully and I'll remove the preferred from 36.2.

**Comment by vvolkl** [comment]
I'm getting a cmake config failure due to a missing pytest, likely because py-nose seems to have been replaced by py-pytest as of v36r2: https://gitlab.cern.ch/gaudi/Gaudi/-/commit/67c95e45fd5e9e126263dd1272bbe8f58c1699e4

**Comment by wdconinc** [comment]
With the latest change (adding dependency on pytest for 36.2:) and even with `+optional` there are no build failures.

@vvolkl Are you running with `--test=root` because even with version 36.0 currently in the builtin repository I get the same ~30/277 failed tests. I'm not thinking those are caused by these new versions.

**Comment by wdconinc** [comment]
ci/gitlab-ci pipeline failed with "ERROR: Job failed (system failure): prepare environment: timed out waiting for pod to start", unrelated to this PR.

**Comment by alalazo** [comment]
@wdconinc Can you rebase this PR to run the pipeline on the latest merge commit?

**Comment by vvolkl** [comment]
I've been taking a look at the test failures, they are indeed unrelated to the versions. There are some test dependencies missing in the recipe, I'll add them in the future PR; this is good to be merged.

**Comment by wdconinc** [comment]
Do you think 36.3 can just be the preferred? Works for me.

**Comment by vvolkl** [comment]
Yes, no reason to stick to an older version!

**Review by alalazo — DISMISSED** [review]

**Comment by wdconinc** [comment]
@alalazo Is there a way to restart the gitlab-ci job that seems to have gone AWOL?

**Comment by alalazo** [comment]
@spackbot run pipeline

**Comment by drbenmorgan** [comment]
@wdconinc, @alalazo, I'm not sure of the cause of the only failure in the CI here: https://gitlab.spack.io/spack/spack/-/jobs/1662779#L1945

The message is obvious about Python 3.6.15 being deprecated - but not sure why that pops up or how to resolve. In any case, the changes to the package are fine, so will approve pending info on this issue, so good to merge if no fixes are needed here.

**Review by drbenmorgan — DISMISSED** [review]

**Comment by alalazo** [comment]
@wdconinc Can you please rebase on the latest `develop`? Or I can do that if it is fine with you that I push to the PR branch. (Rebasing is needed to have Gitlab CI run on a new merge commit)

**Comment by wdconinc** [comment]
Done.

**Comment by vvolkl** [comment]
Now you could even add v36r4 ;). I can add this in a separate PR though.

**Comment by wdconinc** [comment]
Yeah, I was going to suggest a separate PR too.  (Disadvantage of the gaudi repo being hosted on the CERN gitlab server: can't subscribe to release notifications.)

**Review by alalazo — APPROVED** [review]

**Pull request merged**
